conformance/lang/expressions/multiplicative-expr/division.balt

```diff
@@ -0,0 +1,4506 @@
+
+     a = ();
+     b = -float:Infinity;
+     c = a / b;
+     io:println(c.toBalString()); // @output ()
+}
+
+Test-Case: output
+Description: Test division by NaN for different float values.
+Labels: multiplicative-expr, numeric-literal, float-point-literal
+
+public function main() {
+     float a = 1321.32f;
+     float b = float:NaN;
+     float c = a / b;
+     io:println(c); // @output NaN
+     io:println(1321.32f / float:NaN); // @output NaN
+     io:println(b / a); // @output NaN
+     io:println(float:NaN / 1321.32f); // @output NaN
+
+     a = float:NaN;
+     b = float:NaN;
+     c = a / b;
+     io:println(c); // @output NaN
+     io:println(a / b); // @output NaN
+
+     a = float:Infinity;
+     b = float:NaN;
+     c = a / b;
+     io:println(c); // @output NaN
+     io:println(a / b); // @output NaN
+
+     a = -float:Infinity;
+     b = float:NaN;
+     c = a / b;
+     io:println(c); // @output NaN
+     io:println(a / b); // @output NaN
+}
+
+Test-Case: output
+Description: Test division by NaN for different nillable float values.
+Labels: multiplicative-expr, numeric-literal, float-point-literal
+
+public function main() {
+     float? a = 1321.32f;
+     float? b = float:NaN;
+     float? c = a / b;
+     io:println(c); // @output NaN
+
+     a = float:NaN;
+     b = float:NaN;
+     c = a / b;
+     io:println(c); // @output NaN
+
+     a = float:Infinity;
+     b = float:NaN;
+     c = a / b;
+     io:println(c); // @output NaN
+
+     a = -float:Infinity;
+     b = float:NaN;
+     c = a / b;
+     io:println(c); // @output NaN
+
+     a = ();
+     b = float:NaN;
+     c = a / b;
+     io:println(c.toBalString()); // @output ()
+}
+
+Test-Case: output
+Description: Test float division scenarios for user-defined subtypes of float.
+Labels: multiplicative-expr, numeric-literal, float-literal, module-type-defn
+
+type Floats -2.0f|-1.0f|0.0f|1.0f|2.0f;
+
+public function main() {
+    Floats a = 1.0;
+    Floats b = 2.0;
+    float c = a / a;
+    float d = a / b;
+
+    io:println(c); // @output 1.0
+    io:println(a / a); // @output 1.0
+    io:println(d); //@output 0.5
+    io:println(a / b); // @output 0.5
+
+    io:println(b / a); // @output 2.0
+    io:println(b / b); // @output 1.0
+
+    a = -0.0;
+    b = +0.0;
+    Floats e = 2.0f;
+    io:println(e / a); // @output -Infinity
+    io:println(e / b); // @output Infinity
+}
+
+Test-Case: output
+Description: Test decimal division scenarios for user-defined subtypes of decimal.
+Labels: multiplicative-expr, numeric-literal, decimal-literal, module-type-defn
+
+type Decimals -2.0d|-1.0d|0.0d|1.0d|2.0d;
+
+public function main() {
+    Decimals a = 1.0d;
+    Decimals b = 2.0d;
+    decimal c = a / a;
+    decimal d = a / b;
+
+    io:println(a / a); // @output 1.0
+    io:println(c); // @output 1.0
+    io:println(a / b); // @output 0.5
+    io:println(d); // @output 0.5
+
+    io:println(b / a); // @output 2.0
+    io:println(b / b); // @output 1.0
+}
```

Comment:
Shall we add tests for nil lifted scenarios with user-defined subtypes of float and decimal as well?
And also let's check division at the module level too.